homework-submission/homework w1/app.js

```diff
@@ -0,0 +1,114 @@
+const ListItem = ({ description, deadLine, done }) => {
+  return <li className={done}>{description + ', ' + deadLine}</li>;
+};
+
+const Button = ({ text, type, event }) => {
+  return (
+    <button type={type} onClick={event}>
+      {text}
+    </button>
+  );
+};
+
+class App extends React.Component {
+  state = {
+    toDoList: [
+      {
+        id: 1,
+        description: 'Get out of bed',
+        deadLine: '2017-09-11',
+        done: true,
+      },
+      {
+        id: 2,
+        description: 'Brush teeth',
+        deadLine: '2017-09-10',
+        done: false,
+      },
+      {
+        id: 3,
+        description: 'Eat breakfast',
+        deadLine: '2017-09-09',
+        done: false,
+      },
+    ],
+    description: '',
+    deadLine: '',
+    toRemove: null,
+  };
+
+  inputChanged(event) {
+    event.preventDefault();
+    this.setState({ [event.target.name]: event.target.value });
+  }
+
+  addToList(event) {
+    event.preventDefault();
+    const newTodoList = {
+      description: this.state.description,
+      deadLine: this.state.deadLine,
+      done: false,
+    };
+    let array = this.state.toDoList;
+    array.push(newTodoList);
```

Comment:
instead of `array.push` you can create a new array
```
const todoList = [...this.state.toDoList, newTodoList]
```

Reply:
this way is much better thanks a lot!